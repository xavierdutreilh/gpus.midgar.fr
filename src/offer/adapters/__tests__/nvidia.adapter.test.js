jest.mock("node-fetch")

const fetch = require("node-fetch")
const fs = require("fs").promises
const { Response } = jest.requireActual("node-fetch")

const nvidia = require("../nvidia.adapter")

describe("getOffers", () => {
  beforeAll(async () => {
    fetch.mockResolvedValue(
      new Response(
        await fs.readFile(`${__dirname}/mocks/get-nvidia-offers.json`, "utf8")
      )
    )
  })

  it("retrieves nvidia offers", async () => {
    const offers = await nvidia.getOffers()
    expect(fetch).toHaveBeenCalledTimes(1)
    expect(fetch).toHaveBeenCalledWith(
      "https://api.nvidia.partners/edge/product/search?page=1&limit=100&locale=fr-fr&manufacturer=NVIDIA&gpu=RTX%203070%2CRTX%203080%2CRTX%203090&category=GPU"
    )
    expect(offers).toMatchSnapshot()
  })
})
