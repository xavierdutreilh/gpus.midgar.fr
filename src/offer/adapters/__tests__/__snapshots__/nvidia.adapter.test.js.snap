// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getOffers retrieves nvidia offers 1`] = `
Array [
  Object {
    "name": "NVIDIA GEFORCE RTX 3090",
    "status": "upcoming",
    "store": "nvidia",
  },
  Object {
    "name": "ASUS TUF-RTX3090-24G-GAMING",
    "status": "check_availability",
    "store": "nvidia",
  },
]
`;
